collect_app/src/main/java/org/odk/collect/android/utilities/WebCredentialsUtils.java

```diff
@@ -0,0 +1,91 @@
+package org.odk.collect.android.utilities;
+
+import android.net.Uri;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.odk.collect.android.http.HttpCredentials;
+import org.odk.collect.android.http.HttpCredentialsInterface;
+import org.odk.collect.android.preferences.GeneralSharedPreferences;
+import org.odk.collect.android.preferences.PreferenceKeys;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+public class WebCredentialsUtils {
+
+    private static WebCredentialsUtils instance;
+
+    private final Map<String, HttpCredentialsInterface> hostCredentials = new HashMap<>();
+
+    private WebCredentialsUtils() {}
+
+    /**
+     * A factory method to return a new WebCredentialsUtils object.
+     *
+     * @return an instance of WebCredentialsUtils
+     */
+    public static synchronized WebCredentialsUtils getInstance() {
```

Comment:
Dagger is our friend :dagger:

Reply:
I agree @jd-alexander but I always seem to have an issue with @Singleton - Dagger always seems to create a new instance rather than passing back the previously created instance. Do you have any pointers or examples of using @Singleton?